File: Problems/Leetcode/Leetcode/11_container_maxwater.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution
{
public:
    int maxArea(vector<int> &h)
    {
        int l = 0;
        int r = h.size() - 1;

        int max_area = 0;
        while (l < r)
        {
            int s = min(h[l], h[r]);
            max_area = max(max_area, s * (r - l));

            if (h[l] < h[r])
                l++;
            else
                r--;
        }
        return max_area;
    }
};

int main()
{
    int n;
    cin >> n;
    vector<int> v;
    for (int i = 0; i < n; i++)
    {
        int temp;
        cin >> temp;
        v.push_back(temp);
    }

    int s = Solution().maxArea(v);
    cout << s;

    return 0;
}

File: Problems/Leetcode/Leetcode/11_container_maxwater.py

from sys import maxsize


class Solution:
    def maxArea(self, height):
        l = 0
        r = len(height) - 1
        max_area = 0

        while(l < r):
            shorter = min(height[l], height[r])
            temp = shorter*(r-l)
            max_area = max(max_area, temp)

            if(height[l] < height[r]):
                l += 1
            else:
                r -= 1

        return max_area


if __name__ == '__main__':
    l = [int(i) for i in input().split()]
    s = Solution().maxArea(l)
    print(s)
